fairlearn/postprocessing/_threshold_optimizer.py

```diff
@@ -34,44 +36,43 @@
     "was detected. Please rename your column and try again.".format(SCORE_KEY, LABEL_KEY)
 SCORES_DATA_TOO_MANY_COLUMNS_ERROR_MESSAGE = "The provided scores data contains multiple columns."
 UNEXPECTED_DATA_TYPE_ERROR_MESSAGE = "Unexpected data type {} encountered."
-EITHER_PREDICTOR_OR_ESTIMATOR_ERROR_MESSAGE = "Only one of 'unconstrained_predictor' and " \
-                                              "'estimator' can be passed."
-PREDICTOR_OR_ESTIMATOR_REQUIRED_ERROR_MESSAGE = "One of 'unconstrained_predictor' and " \
-                                                "'estimator' need to be passed."
-
+ESTIMATOR_ERROR_MESSAGE = "'estimator' cannot be `None`."
 
 _SUPPORTED_CONSTRAINTS = [DEMOGRAPHIC_PARITY, EQUALIZED_ODDS]
 
 logger = logging.getLogger(__name__)
 
 
-class ThresholdOptimizer(ClassifierMixin, BaseEstimator):
+class ThresholdOptimizer(BaseEstimator, ClassifierMixin, MetaEstimatorMixin):
     """An Estimator based on the threshold optimization approach.
 
     The procedure followed is described in detail in
     `Hardt et al. (2016) <https://arxiv.org/abs/1610.02413>`_.
 
-    :param unconstrained_predictor: The trained predictor whose output will be post processed
-    :type unconstrained_predictor: A trained predictor
-    :param estimator: An untrained estimator that will be trained, and
-        subsequently its output will be post processed
-    :type estimator: An untrained estimator
+    :param estimator: An estimator whose output will be post processed
+    :type estimator: An estimator
     :param grid_size: The number of ticks on the grid over which we evaluate the curves.
         A large grid_size means that we approximate the actual curve, so it increases the chance
         of being very close to the actual best solution.
     :type grid_size: int
     :param flip: Allow flipping to negative weights if it improves accuracy.
     :type flip: bool
+    :param refit: If ``False``, avoid re-fitting the given estimator if it's
+        already trained. Note that when used with ``cross_val_score``,
+        ``GridSearchCV`` and similar utilities that clone the estimator,
+        the effective behavior is ``refit=True``.
+    :type: bool, default=True
     """
 
-    def __init__(self, *, unconstrained_predictor=None, estimator=None,
-                 constraints=DEMOGRAPHIC_PARITY, grid_size=1000, flip=True):
+    def __init__(self, *, estimator=None,
+                 constraints=DEMOGRAPHIC_PARITY, grid_size=1000, flip=True,
+                 refit=True):
         self.grid_size = grid_size
         self.flip = flip
         self.post_processed_predictor_by_sensitive_feature = None
         self.constraints = constraints
-        self.unconstrained_predictor = unconstrained_predictor
         self.estimator = estimator
+        self.refit = refit
 
```

Comment:
I have a slight preference for `prefit` pattern (`prefit=True` to indicate when things should not be refitted) over the `refit` pattern.

The main reason is that `prefit` pattern already exists in [SelectFromModel](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.SelectFromModel.html#sklearn-feature-selection-selectfrommodel) and somewhat in [CallibratedClassifierCV](https://scikit-learn.org/stable/modules/generated/sklearn.calibration.CalibratedClassifierCV.html#sklearn-calibration-calibratedclassifiercv). On the other hand, the usage of `refit` in [GridSearchCV](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html#sklearn-model-selection-gridsearchcv) is a bit different.